### DISCUSSION TITLE
autodiscoverFilter not excluding multiple repos

### DISCUSSION THREAD

**Discussion by emillaine** [renovatebot/renovate#19298] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

28.16

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.4.5

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I want to use autodiscover, but exclude two repos, `org-a/repo-a` and `org-b/repo-b`.

I am setting `RENOVATE_AUTODISCOVER_FILTER` environment variable as `"!(org-a/repo-a|org-b/repo-b)"` (as per minimatch docs).

Renovate logs `"autodiscoverFilter": "!(org-a/repo-a|org-b/repo-b)"` as a value in the config, indicating that the env var was used.

Autodiscover still processes these repos: Renovate logs:

```
DEBUG: Autodiscovering GitHub repositories
INFO: Autodiscovered repositories
      "length": 79,
      "repositories": [
        ...
        "org-a/repo-a",
        ...
        "org-b/repo-b",
        ...
      ]
```

And the repos are processed as without autodiscoverFilter.

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
Please test it using minimatch like this: https://runkit.com/embed/ep80o75pnkl8

**Reply by emillaine** [reply]
That is exactly what I used. And as can be seen in your runkit, it doesn't work.

I also tried using a negated regex as per [Renovate docs](https://docs.renovatebot.com/self-hosted-configuration/#autodiscoverfilter): `"!/online-web/epic-ui|social-ecosystem/social-overlay-app/"`. It didn't work either (same outcome).

**Reply by emillaine** [reply]
Created https://github.com/isaacs/minimatch/issues/186 for the minimatch bug.

**Reply by rarkins** [reply]
If you're using regex, you need to escape forward slashes, e.g. `"!/online-web\/epic-ui|social-ecosystem\/social-overlay-app/"`

**Reply by emillaine** [reply]
The docs don't escape the forward slashes in regex: https://docs.renovatebot.com/self-hosted-configuration/#autodiscoverfilter

**Reply by emillaine** [reply]
Sorry I put the wrong version number in the post, I'm actually using Renovate v28.16 which doesn't support regex autodiscoverFilter

**Reply by rarkins** [reply]
It seems the regex matching works with escaped or non escaped, so stick with the non-escaped as shown in the docs

**Reply by viceice** [reply]
i would prefer to stick to escaped slashes, as the other  way it's pure lazy regex engine behavior and can break in future.

**Reply by emillaine** [reply]
The solution that worked for me was: `'!{org-a/repo-a,org-b/repo-b}'` https://github.com/isaacs/minimatch/issues/186#issuecomment-1342216214
